src/bidiMapper/domains/network/NetworkProcessor.ts

```diff
@@ -75,8 +75,30 @@ export class NetworkProcessor {
     throw new UnknownCommandException('Not implemented yet.');
   }
 
-  failRequest(_params: Network.FailRequestParameters): EmptyResult {
-    throw new UnknownCommandException('Not implemented yet.');
+  async failRequest(
+    params: Network.FailRequestParameters
+  ): Promise<EmptyResult> {
+    const networkId = params.request;
+    const blockedRequest = this.#networkStorage.getBlockedRequest(networkId);
+    const {request: fetchId, phase} = blockedRequest;
+
+    if (phase === Network.InterceptPhase.AuthRequired) {
+      throw new InvalidArgumentException(
+        `Blocked request for network id '${networkId}' is in 'AuthRequired' phase`
+      );
+    }
+
+    const cdpTarget = this.#networkStorage.getRequest(networkId)?.cdpTarget;
```

Comment:
I suggest moving this into a method on NetwrokRequest

Reply:
Done